ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/l3_interfaces.py

```diff
@@ -64,3 +66,50 @@ def l3_interfaces(self: SharedUtils) -> list:
             l3_interfaces = [self.apply_l3_interfaces_profile(l3_interface) for l3_interface in l3_interfaces]
 
         return l3_interfaces
+
+    @cached_property
+    def l3_interfaces_bgp_neighbors(self: SharedUtils) -> list:
+        neighbors = []
+        for interface in self.l3_interfaces:
+            peer_ip = get(interface, "peer_ip")
+            bgp = get(interface, "bgp")
+            if not (peer_ip and bgp):
+                continue
+
+            peer_as = get(bgp, "peer_as")
+            if peer_as is None:
+                raise AristaAvdMissingVariableError(f"'l3_interfaces[{interface['name']}].bgp.peer_as' needs to be set to enable BGP.")
+
+            prefixes = get(bgp, "prefixes_allowed_in")
+            if prefixes is None:
+                raise AristaAvdMissingVariableError(f"BGP is enabled but 'bgp.prefixes_allowed_in' is not configured for l3_interfaces[{interface['name']}]")
```

Comment:
This check should only be for wan interfaces. So if a wan_carrier has been set. (I don't remember if we maintain a list of wan interface names somewhere that would be easy to check)

Reply:
We can call wan_interfaces function, but easier to check if carrier is set or not.